app/controllers/spree/checkout_controller_decorator.rb

```diff
@@ -30,8 +30,4 @@ def before_address
     @order.bill_address ||= preferred_bill_address || last_used_bill_address || Spree::Address.default
     @order.ship_address ||= preferred_ship_address || last_used_ship_address || nil
   end
-
-  def after_complete
-    reset_order
-  end
```

Comment:
Awesome, thanks so much. That is much clearer. My fault: I missed your explanation in the commit message, sorry!

My one qualm is having `expire_current_order ` defined multiple times, and it feels a little disjointed calling a controller method directly from a service class, but I can't see a clear solution to either of those things, so I think this is good to go. Test coverage looks great, and the encapsulation of logic is 👌 

Given that this is not a trivial change, I would like to push it through the build pipeline - that has historically been how everything is merged into `master`. Sally isn't available to test tomorrow, so that will mean that it won't be merged until Friday - is that ok? - or does this need to be higher priority? If sol, I can give a quick look tomorrow.